### PR TITLE
PoseidonBranch minimum depth fixed #112

### LINKED ISSUE #112: PoseidonTree - Merkle opening failure
After the NStack upgrade to latest, the merkle opening proof is failing for the transfer contract.

Test case: Execute circuit with 2 inputs and 2 outputs

### PR DESCRIPTION
**Pull request by vlopes11**
Fixes #112

### CODE REVIEW

File: src/sponge/mod.rs [hunk 1]
```diff
@@ -4,6 +4,8 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
-pub mod sponge;
+mod sponge;
 
+#[cfg(feature = "std")]
```
Comment (CPerezz) [inline src/sponge/mod.rs:9]: Why is `std` instead of `alloc` the feature for this fn?

File: src/tree/branch.rs [hunk 2]
```diff
@@ -42,33 +45,45 @@ impl AsRef<[BlsScalar]> for PoseidonLevel {
 }
 
 /// Represents a full path for a merkle opening
-#[derive(Debug, Clone, Copy)]
-pub struct PoseidonBranch<const DEPTH: usize>([PoseidonLevel; DEPTH]);
+#[derive(Debug, Clone)]
+pub struct PoseidonBranch<const DEPTH: usize> {
+    path: Vec<PoseidonLevel>,
+}
```
Comment (CPerezz) [inline src/tree/branch.rs:51]: I guess that const generics just constraint the type here to disseminate between `PoseidonBranch`es of different depths. Is that the case?

Comment (CPerezz) [reply src/tree/branch.rs:51]: Also, I see that you didn't go for the Boxed array and the root attrb. At the end it;s the same. But would be nice to add some docs on how the leafs/levels are positioned in the `PoseidonBranch`

**Review by CPerezz — CHANGES_REQUESTED** [review]
A few questions and requests

File: src/tree/branch.rs (hunk 2 above)
Comment (vlopes11) [reply src/tree/branch.rs:51]: The idea was to mirror the implementation we have for kelvin so we have a consistent allocator across the stack.
https://github.com/dusk-network/microkelvin/blob/main/src/branch.rs#L88

The `root` attribute can be created, but to remove it from the path would require some unnecessary refactoring. I believe it makes more sense to have an iterator over the path that just ends up with the root. By separating one from the other, we will need to increase the code complexity a lot.

The `Vec` solves the unstable `DEPTH + 1` approach so we don't really have anything to restrict this implementation

File: src/sponge/mod.rs (hunk 1 above)
Comment (vlopes11) [reply src/sponge/mod.rs:9]: This is related to the plonk requirements. Plonk is defined under `std` while `Vec` is defined under `alloc`

Comment (CPerezz) [reply src/sponge/mod.rs:9]: We need to work on that at some point. Port plonk to no_std using alloc and vec. Otherways this is non-sensical...

File: src/tree/branch.rs (hunk 2 above)
Comment (CPerezz) [reply src/tree/branch.rs:51]: Makes sense. If the rest are OK keeping the const-generics to constraint the type, for me it's ok.

**Review by CPerezz — APPROVED** [review]
LGTM after the discussions with @vlopes11

**Review by krl — APPROVED** [review]
I will approve of this, but i'm getting less certain that this type is actually needed at all.

File: src/sponge/mod.rs (hunk 1 above)
Comment (vlopes11) [reply src/sponge/mod.rs:9]: Cool, we can register an issue to address this later on

**Pull request merged**